### PR TITLE
PerfCompare app

### PR DESCRIPTION
**Pull request by jonthysell**
PerfCompare provides a single app to compare the performance of loading a simple chat-window mockup UI across XAML Markup, RNW Current, and RNW VNext.

Covers issues #2127, #2374, #2375, #2377, #2378, #2376, #2452

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/2512)

### CODE REVIEW

File: vnext/include/ReactUWP/IReactInstance.h [hunk 1]
```diff
@@ -71,8 +72,14 @@ struct IReactInstance
   virtual const std::string& LastErrorMessage() const noexcept = 0;
 
   virtual void loadBundle(std::string&& jsBundleRelativePath) = 0;
-};
 
-void UpdateDevSettings(bool useWebDebugger, bool useLiveReload, bool reuseReactInstances);
+  // Test Hooks
+  virtual void SetTestHook(std::string&& testHookName, std::function<void()> testHook) = 0;
```
Comment (ahimberg) [inline vnext/include/ReactUWP/IReactInstance.h]: Is there some other way to do this than adding as much overhead to ReactInstance?  Listen for an ETW event, use a native module, or take a raw pointer with only one function signature so at least the instance can bail quickly.  And not using a std::string, just a const char*.

Comment (jonthysell) [reply vnext/include/ReactUWP/IReactInstance.h]: There may be other ways for other needs, but at some point we're going to need to be able to hook into the creation/modification of XamlViews at various times and perform checks on them from within native integration tests. Right now our only option is to search the tree after the fact.

Comment (ahimberg) [reply vnext/include/ReactUWP/IReactInstance.h]: Given what we're doing now I would probably just go with a single test hook that can be set, which is a callback, then there is just one set function and the call test hook can be a simple null check.  If we're doing more seeing how we can use ETL and the systrace statements throughout the RN C++ and Js code seems like the proper way to trace perf.

Comment (jonthysell) [reply vnext/include/ReactUWP/IReactInstance.h]: Ok, I've minimized the test hooks.

**Comment by jonthysell** [comment]
@ahimberg: Can you look at the updated test hook surface?

**Comment by ahimberg** [comment]
You might want to use just RunAsync at normal pri instead of idle to get more reliable numbers (?)

**Pull request merged**